internal/policy/verify_test.go

```diff
@@ -81,6 +86,64 @@ func TestVerifyRelativeForRef(t *testing.T) {
 	assert.ErrorIs(t, err, rsl.ErrRSLEntryNotFound)
 }
 
+func TestVerifyCommit(t *testing.T) {
+	repo, _ := createTestRepository(t, createTestStateWithPolicy)
+	refName := "refs/heads/main"
+	gpgKeyBytes, err := os.ReadFile(filepath.Join("test-data", "gpg-pubkey.asc"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	gpgKey, err := gpg.LoadGPGKeyFromBytes(gpgKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs := common.AddNTestCommitsToSpecifiedRef(t, repo, refName, 3)
+	entry := rsl.NewEntry(refName, commitIDs[len(commitIDs)-1])
+	entryID := common.CreateTestRSLEntryCommit(t, repo, entry)
+	entry.ID = entryID
+
+	expectedStatus := make(map[string]string, len(commitIDs))
+	commitIDStrings := make([]string, 0, len(commitIDs))
+	for _, c := range commitIDs {
+		commitIDStrings = append(commitIDStrings, c.String())
+		expectedStatus[c.String()] = fmt.Sprintf(goodSignatureMessageFmt, gpgKey.KeyType, gpgKey.KeyID)
+	}
+
+	// Verify all commit signatures
+	status := VerifyCommit(context.Background(), repo, commitIDStrings)
+	assert.Equal(t, expectedStatus, status)
+
+	if err := repo.Storer.SetReference(plumbing.NewSymbolicReference(plumbing.HEAD, plumbing.ReferenceName(refName))); err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify signature for HEAD and branch
+	expectedStatus = map[string]string{
+		"HEAD":  fmt.Sprintf(goodSignatureMessageFmt, gpgKey.KeyType, gpgKey.KeyID),
+		refName: fmt.Sprintf(goodSignatureMessageFmt, gpgKey.KeyType, gpgKey.KeyID),
+	}
+	status = VerifyCommit(context.Background(), repo, []string{"HEAD", refName})
+	assert.Equal(t, expectedStatus, status)
+
+	// Try a tag
+	tagHash, err := gitinterface.Tag(repo, commitIDs[len(commitIDs)-1], "v1", "Test tag", false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedStatus = map[string]string{tagHash.String(): nonCommitMessage}
+	status = VerifyCommit(context.Background(), repo, []string{tagHash.String()})
```

Comment:
nit: define the ctx at the top of the test and reuse (this makes it easier to add context info later if needed)